test/test_alias.cpp

```diff
@@ -368,6 +368,50 @@ TEST_F(AliasTest, DuplicateOutputs) {
       __FILE__);
 }
 
+TEST_F(AliasTest, SliceToSizeOne_Issue1353) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({4, 6, 7});
+  fusion->addInput(in);
+  TensorView* out = slice(in, {0, 0, 0}, {4, 6, 1});
+  fusion->addOutput(out);
+
+  FusionExecutorCache fec(std::move(fusion));
+  at::Tensor in_tensor = at::randn({4, 6, 7}).cuda();
+  at::Tensor out_tensor = fec.runFusionWithInputs({in_tensor})[0];
+  EXPECT_TRUE(out_tensor.is_alias_of(in_tensor));
+
+  testValidate(
+      fec.fusion(),
+      {in_tensor.slice(/*dim=*/2, /*start=*/c10::nullopt, /*end=*/1)},
+      {in_tensor},
+      __LINE__,
+      __FILE__);
+}
+
+TEST_F(AliasTest, SliceRightOfBroadcast) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({4, 1, 7});
+  fusion->addInput(in);
+  TensorView* out = slice(in, {0, 0, 0}, {4, 1, 5});
+  fusion->addOutput(out);
+
+  FusionExecutorCache fec(std::move(fusion));
+  at::Tensor in_tensor = at::randn({4, 1, 7}).cuda();
+  at::Tensor out_tensor = fec.runFusionWithInputs({in_tensor})[0];
+  EXPECT_TRUE(out_tensor.is_alias_of(in_tensor));
```

Comment:
You would know better than me, but this confused me. I'm assuming that `7` is the fastest dimension in `{4,1,7}`, and it might be that I just have this backwards :-)

But if I have it the right way, I think of two tensors as aliases when they use the same base address for the data, but wouldn't the `out` tensor need a stride of `7` (despite the dim being `5`) in the fastest dim for that to be true? Does the alias analysis / usage perhaps guarantee that `out` indeed *does* get a stride `7`? If not, where did my logic go awry?

Reply:
Think about this in the python indexing. Any basic indexing (slice included) can be done via a meta update as a `view` into the original storage and you don't need copy.

after the slice, the stride on each dimension remains the same and we are only updating the shape (and offset when it applies)

Reply:
```
import numpy as np
arr = np.arange(28).reshape(4, 1, 7)
b = arr[:,:,0:5]
```

```
>>> arr.strides
(56, 56, 8)
>>> b.strides
(56, 56, 8)
>>> arr.ctypes.data
94247211410336
>>> b.ctypes.data
94247211410336
```

Reply:
> I'm assuming that 7 is the fastest dimension in {4,1,7}

Right. 

> I think of two tensors as aliases when they use the same base address for the data

No, they alias when they share the storage buffer but the base addresses don't have to be the same. I believe @jjsjann123 answered this question. See also `AliasTest.SliceViewPermute` where all the three outputs are alias but only the first one uses the same base address.